sale_import_base/components/importer_sale_channel.py

```diff
@@ -0,0 +1,339 @@
+#  Copyright (c) Akretion 2020
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from copy import deepcopy
+
+from marshmallow_objects import ValidationError as MarshmallowValidationError
+
+from odoo.exceptions import ValidationError
+
+from odoo.addons.component.core import Component
+
+MAPPINGS_SALE_ORDER_ADDRESS_SIMPLE = [
+    "name",
+    "street",
+    "street2",
+    "zip",
+    "city",
+    "email",
+]
+
+
+class ImporterSaleChannel(Component):
+    _inherit = "processor"
+    _name = "importer.sale.channel"
+    _apply_on = ["sale.order"]
+    _usage = "json_import"
+
+    def run(self, raw_data):
+        """
+        :param raw_data: json-like string
+        :return: generated sale order
+        """
+        # we need this step because the chunk data is stored as
+        # a string (so we can edit it)
+        try:
+            so_datamodel_load = self.env.datamodels["sale.order"].load_json(raw_data)
+        except MarshmallowValidationError:
+            raise ValidationError(MarshmallowValidationError)
+        except Exception:
+            raise
+        json_data_initial = so_datamodel_load.dump()
+        json_data = deepcopy(json_data_initial)
+        so_vals = self._si_process_data(json_data)
+        new_sale_order = self.env["sale.order"].create(so_vals)
+        self._si_finalize(new_sale_order, json_data_initial)
+        return new_sale_order
+
+    # DATAMODEL PROCESSORS
+    def _si_process_data(self, so_vals):
+        """ Transform values in-place
+         to make it usable in create() """
+        self._si_process_simple_fields(so_vals)
+        self._si_process_m2os(so_vals)
+        so_vals = self._si_simulate_onchanges(so_vals)
+        return so_vals
+
+    def _si_process_simple_fields(self, so_vals):
+        del so_vals["currency_code"]
+        del so_vals["payment"]
+
+    def _si_process_m2os(self, so_vals):
+        channel_id = self._si_get_channel(so_vals)
```

Comment:
we can have the channel by doing self.collection.reference (as the queue.job.chunk is link to the channel)